contracts/Splitter.sol

```diff
@@ -7,34 +7,29 @@ contract Splitter {
     
     address payable[] internal recipients = [0x20601F1Ddb44E28b1511EdBD316A368D80116408, //"Bob";
                                             0x24F1500890505ceD8534502ab403F20ce99feea0]; //"Carol";
+    uint InitialValue = 100 ether;
 
     event Deposit(uint date,
                   address indexed _from,
                   address indexed _to,
                   uint amount);
 
     function deposit() external payable{
-        doDeposit(msg.sender, msg.value);
-    }
 
-    function doDeposit(address _sender, uint _value) public payable {
-        if(_sender != sender){
+        if(msg.value % 2 != 0) {
+            revert("Only even numbers can be sent");
+        }
+
+        if(msg.sender != sender){
             revert("Only Alice can send ether");
         }
-        
-        uint split = split(_value);
-        
+
+        uint split = msg.value / 2;
+
         for (uint i=0; i < recipients.length; i++) {
             recipients[i].transfer(split);
```

Comment:
There is a security concern here but I will set it aside for now. Let's nudge this in the right direction first and see if the concern is still there. Remind me to unfold it for you before we move on but don't worry about it for now.

Reply:
Will do.